tools/clang_tidy.py

```diff
@@ -41,27 +43,27 @@
 
 # @@ -start,count +start,count @@
 CHUNK_PATTERN = r"^@@\s+-\d+(?:,\d+)?\s+\+(\d+)(?:,(\d+))?\s+@@"
+CLANG_WARNING_PATTERN=re.compile("([^:]+):(\d+):\d+:\s+warning:.*\[([a-z\-,]+)\]")
```

Comment:
I think most of the Flake8 warnings here are because you're not using a raw string:
```suggestion
CLANG_WARNING_PATTERN = re.compile(r"([^:]+):(\d+):\d+:\s+warning:.*\[([a-z\-,]+)\]")
```